From 7e042e4e32696978e6f3e83b516370a39a1b8b6f Mon Sep 17 00:00:00 2001
Message-ID: <7e042e4e32696978e6f3e83b516370a39a1b8b6f.1694125610.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1694125609.git.sharlatanus@gmail.com>
References: <cover.1694125609.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Thu, 7 Sep 2023 21:41:39 +0100
Subject: [PATCH 15/19] gnu: python-roman-datamodels: Update to 0.17.1.

* gnu/packages/astronomy.scm (python-roman-datamodels): Update to
0.17.1.
[arguments]{phases}: Add 'set-env phase to pass tests.
[native-inputs]: Remove python-pytest-openfiles. Add python-pytest-env.
---
 gnu/packages/astronomy.scm | 17 ++++++++++++-----
 1 file changed, 12 insertions(+), 5 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index a402c63e9f..d1b34def8f 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -3835,13 +3835,13 @@ (define-public python-rad
 (define-public python-roman-datamodels
   (package
     (name "python-roman-datamodels")
-    (version "0.15.0")
+    (version "0.17.1")
     (source (origin
               (method url-fetch)
               (uri (pypi-uri "roman_datamodels" version))
               (sha256
                (base32
-                "0frhm1cqqd8934yizhm4fy78y38q2w9ncm4rv1n74hfypkyis4ap"))))
+                "1y12cp8172i4a314gmhpi86jw6pfylz1adh0rzr5zqmvd3mrjqlj"))))
     (build-system pyproject-build-system)
     (arguments
      ;; XXX: Check how to make all tests enabled, probably some more inner
@@ -3850,7 +3850,12 @@ (define-public python-roman-datamodels
      (list #:test-flags #~(list "-k"
                                 (string-append "not test_will_validate"
                                  " and not test_will_strict_validate"
-                                 " and not test_nuke_validation"))))
+                                 " and not test_nuke_validation"))
+      #:phases
+      #~(modify-phases %standard-phases
+          (add-after 'unpack 'set-env
+            (lambda _
+              (setenv "HOME" "/tmp"))))))
     (propagated-inputs (list python-asdf
                              python-asdf-astropy
                              python-asdf-standard
@@ -3859,8 +3864,10 @@ (define-public python-roman-datamodels
                              python-numpy
                              python-psutil
                              python-rad))
-    (native-inputs (list python-pytest python-pytest-doctestplus
-                         python-pytest-openfiles python-semantic-version
+    (native-inputs (list python-pytest
+                         python-pytest-doctestplus
+                         python-pytest-env
+                         python-semantic-version
                          python-setuptools-scm))
     (home-page "https://github.com/spacetelescope/roman_datamodels")
     (synopsis "Roman Datamodels Support")
-- 
2.41.0

